Sources/xcodeproj/Extensions/AbsolutePath+Extras.swift

```diff
@@ -39,7 +39,19 @@ extension AbsolutePath {
     /// - Parameter content: content to be written.
     /// - Throws: an error if the writing fails.
     func write(_ content: String) throws {
-        try content.write(toFile: asString, atomically: true, encoding: .utf8)
+
+        // Check the current file
+        if exists {
+            // If the new content is the same as the old then don't overwrite.
+            if let oldContents = try? read(), oldContents == content {
```

Comment:
This is the read I was referring to. If the project is large, this might be expensive.

Reply:
Yep. I initially thought of using a MD5 checksum or similar, but haven't found an easy way to do it. Might have to chase it again as we don't care about what the contents are, just whether they have changed or not. The goal being to avoid writing the same content. However if it ends up being no improvement I'll just drop the idea and concentrate on the directory deletion part which I think is what was giving Xcode a problem.